Firestore/Source/Auth/FSTCredentialsProvider.m

```diff
@@ -74,28 +67,31 @@ - (instancetype)initWithApp:(FIRApp *)app {
   self = [super init];
   if (self) {
     _app = app;
-    _auth = [FIRAuth authWithApp:app];
-    _currentUser = [[FSTUser alloc] initWithUID:self.auth.currentUser.uid];
+    _currentUser = [[FSTUser alloc] initWithUID:[self.app getUID]];
     _userCounter = 0;
 
     // Register for user changes so that we can internally track the current user.
     FSTWeakify(self);
-    _authListenerHandle = [self.auth addAuthStateDidChangeListener:^(FIRAuth *auth, FIRUser *user) {
-      FSTStrongify(self);
-      if (self) {
-        @synchronized(self) {
-          FSTUser *newUser = [[FSTUser alloc] initWithUID:user.uid];
-          if (![newUser isEqual:self.currentUser]) {
-            self.currentUser = newUser;
-            self.userCounter++;
-            FSTVoidUserBlock listenerBlock = self.userChangeListener;
-            if (listenerBlock) {
-              listenerBlock(self.currentUser);
-            }
-          }
-        }
-      }
-    }];
+    _authListenerHandle = [[NSNotificationCenter defaultCenter]
+        addObserverForName:FIRAuthStateDidChangeInternalNotification
+                    object:nil
+                     queue:nil
+                usingBlock:^(NSNotification *notification) {
```

Comment:
All instances of Firestore will get these internal notifications from all instances of auth so this needs to pull the app instance out of the notification and only perform this logic if the notification's app instance is the same as the app instance governing this instance of Firestore.

Reply:
Done. See the (currently) last commit in this PR:
https://github.com/firebase/firebase-ios-sdk/pull/505/commits/53db38c41ef6826ed32d6f0c3657c462867a11c7